File: ECommerceApi/Models/Shop.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ECommerceApi.Models
{
    public class Shop
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        [Required]
        public int OwnerId { get; set; }
        public User? Owner { get; set; }

        [Required]
        [MaxLength(100)]
        public string Slug { get; set; } = string.Empty;

        public string? LogoUrl { get; set; }
        public string? BannerUrl { get; set; }

        public string TextColor { get; set; } = "#FFFFFF";
        public string BackgroundColor { get; set; } = "#000000";
        public string ThemeColor { get; set; } = "#2563EB";

        public string? Email { get; set; }
        public string? Phone { get; set; }

        public int ProductCount { get; set; } = 0;

        public bool IsActive { get; set; } = true;

        [NotMapped]
        public double AverageRating { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: ecommerce-frontend/types/shop.ts
// types/shop.ts
import { User } from './user';
import { Product } from './product';

export interface Shop {
  id: number;
  name: string;
  slug: string;
  description: string;
  logoUrl: string | null;
  bannerUrl: string | null;
  themeColor: string;
  backgroundColor: string;
  textColor: string;
  email: string | null;
  phone: string | null;
  productCount: number;
  createdAt: string;
  updatedAt?: string;
  ownerId: number;
  owner?: User;
  products?: Product[];
}

export interface ShopResponse {
  id: number;
  name: string;
  slug: string;
  description: string;
  logoUrl: string | null;
  bannerUrl: string | null;
  productCount: number;
  ownerId: number;
  username?: string;
  // Ces propriétés peuvent venir de l'API ou être optionnelles
  email?: string | null;
  phone?: string | null;
  createdAt?: string;
}

export interface CreateShopRequest {
  name: string;
  description: string;
  themeColor?: string;
  backgroundColor?: string;
  textColor?: string;
  email?: string;
  phone?: string;
}

export interface UpdateShopRequest {
  name?: string;
  description?: string;
  themeColor?: string;
  backgroundColor?: string;
  textColor?: string;
  email?: string;
  phone?: string;
}

export interface ShopListResponse {
  data: ShopResponse[];
  totalPages: number;
  currentPage: number;
  totalCount: number;
}

export const isShopOwner = (userId: number | undefined, shop: Shop | ShopResponse | null): boolean => {
  if (!userId || !shop) return false;
  return shop.ownerId === userId;
};